staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -59,26 +60,40 @@ var (
 
 		# Create a debug container named debugger using a custom automated debugging image.
 		# (requires the EphemeralContainers feature to be enabled in the cluster)
-		kubectl alpha debug --image=myproj/debug-tools -c debugger mypod`))
+		kubectl alpha debug --image=myproj/debug-tools -c debugger mypod
+
+		# Create a debug container in a copy of the original Pod and attach to it
+		kubectl alpha debug mypod -it --image=busybox --copy-to=my-debugger
+
+		# Create a copy of mypod named my-debugger with my-container's image changed to busybox
+		kubectl alpha debug mypod --image=busybox --container=my-container --copy-to=my-debugger -- sleep 1d
```

Comment:
Now that I'm looking at this I found it weird and hard to decipher. Have a look at how we do something similar in `kubectl set image`:
```
# Update image of all containers of daemonset abc to 'nginx:1.9.1'
kubectl set image daemonset abc *=nginx:1.9.1
```
How about supporting something similar here as well. Iow. when I pass `--image=busybox` it applies to all, similarly `--image=*=busybox`. But `--image=my-container=busybox` would apply that image only for that. @aylei @verb wdyt?

Reply:
IMHO, current syntax works better for `debug`. A common use case of `debug` is to add a "debug" container, either by making a copy or adding an ephemeral container, to the pod so that user can diagnose their business containers with the tools installed in the debug container. So usually `debug` does't have to and also shouldn't mutate the image of other business containers. And `--image`, in semantic, is always the container image to use for the **debug container**.

`--container` is only used to specify the name of the debug container before debug by copy is added. When debug by copy, `debug` will change the image of an existing container if `--container` is set to an existing on deliberately, I think it is a not a common use case of `debug` and is only implemented for completeness.

Reply:
There are two debugging scenarios I hoped to address with debug by copy:

1. The same as addressed in debug by ephemeral container, but ephemeral containers are unavailable or the operator prefers to operate on a copy.
2. A primary container in the pod is crashlooping and I want to replace the container image with (for example) busybox so I can examine the environment with a shell.

I suspect it would be uncommon to change multiple or all container images at once. Rather than addressing that directly maybe it'd be better to have an `--edit` that opens an editor prior to create, similar to `kubectl edit`, which would be a useful catch-all for arbitrary changes.

The mutate specification of `kubectl set image` seems more generically useful, but I'm inclined to keep `kubectl debug` focused on specific debugging journeys. It would be easy to add `--image=*=busybox` functionality in the future while remaining backwards compatible with current "only modify a single container" behavior, so I'd omit it for now.

It is a bit hard to decipher, though, and I think it's because it's unclear which flags operate on pod names and which operate on containers. Should we rename `--copy-to` to `--copy-pod-to`?

Reply:
IMHO, `--copy-to` is simpler but still accurate enough, since the object of `debug by copy` is exactly a Pod.

Reply:
The form I'm suggesting is addressing that particular use-case you're talking about @verb. If you have a pod with multiple containers how do you specify which one you replace with the above syntax? With the one I'm proposing it's explicit. I'd prefer not to have `--edit` option, that seems like an overkill. @aylei I agree a Pod's copy, but you are mutating the copy with your own image, and you need a way to say which image and where. As long as you have just one container, you're fine. With multiple it becomes a problem.

Reply:
Ok, SGTM. I don't feel strongly, and being able to mutate multiple images is undoubtedly useful. It's nice to use the same mutation language as other parts of kubectl.

Should we allow `--container` && `--image` mutation syntax at the same time? That is, can we add a new container *and* change the image of the existing ones with a command line:

```
kubectl alpha debug mypod --copy-to=my-debugger --container=new-container --image=new-container=busybox,old-container=debug-image
```

Since `debug` sometimes adds containers and sometimes changes existing containers, it might be less error prone to *require* the mutation syntax like `--image=*=new-image` for changes to existing images rather than have a default mutate action for `--image=new-image`.